### ISSUE TITLE
SC2053 Quote the rhs of = in [[ ]] to prevent glob interpretation.

### ISSUE THREAD

**Issue by grymsten** [koalaman/shellcheck#291]
Hey,
for this script one comment is reported:

> SC2053 Quote the rhs of = in [[ ]] to prevent glob interpretation.

```bash
#!/bin/bash
string1="te st"
string2="te st"
if [[ ${string1} == "te st" ]]; then
  echo "pattern match!"
fi

if [[ ${string1} == ${string2} ]]; then
  echo "pattern match!"
fi
```

But http://mywiki.wooledge.org/BashFAQ/031 says: 
> No WordSplitting or glob expansion will be done for [[ (and therefore many arguments need not be quoted)

..Is the wiki wrong?

**Comment by dualbus** [comment]
> But http://mywiki.wooledge.org/BashFAQ/031 says: 
> > No WordSplitting or glob expansion will be done for [[ (and therefore many arguments need not be quoted)
> 
> ..Is the wiki wrong?
wiki.wooledge.org is a pretty good source of information, but... it's a wiki.
Sometimes idiots add useless information to it. The wiki is wrong. There are
certain cases when quoting matters inside [[

case #1:
 [[ LHS = RHS ]]
          ^- if you don't quote this, it will try to do pattern matching (with
             globs).

|  dualbus@dualbus ~ % bash -c 'string1=yolo string2=*; [[ $string1 = $string2 ]]; echo $?'
|  0
|  dualbus@dualbus ~ % bash -c 'string1=yolo string2=*; [[ $string1 = "$string2" ]]; echo $?'
|  1


case #2:
 [[ STR =~ REGEX ]]
           ^- quoting here is FUBAR. The recommended practice is to:
              re='some.re'; [[ $str =~ $re ]]

case #3:
 Using $* or any other variant.

|   dualbus@dualbus ~ % bash -c 'IFS=:; set -- 1 2 3; [[ 1:2:3 = $* ]]; echo $?'
|   1
|   dualbus@dualbus ~ % bash -c 'IFS=:; set -- 1 2 3; [[ 1:2:3 = "$*" ]]; echo $?'
|   0

These are the ones that I remember right now. There are more, probably.

**Comment by grymsten** [comment]
Aye, thanks for your explanation. I didnt feel certain about the "error". Issue can be closed.

**Comment by DrVanScott** [comment]
Note: in case 1 above it needs to read

    string1=yolo string2=*;

instead of

    string1=yolo string2=;

**Comment by DrVanScott** [comment]
So as a rule of thumb: Is it always recommended to doule quote RHS in [[ ]] ?

**Comment by koalaman** [comment]
You have to quote if you want to compare strings.

You cannot quote if you want to pattern match.
